tests/test_jsonitem.py

```diff
@@ -60,6 +60,16 @@ def bad_array(request):
     ][request.param]
 
 
+@pytest.fixture(params=range(4))
+def bad_url(request):
+    return [
+        'www.test.com',
+        'http:/hi.com',
+        'https//www.hi.com',
+        'http://bad.port:here'
+    ][request.param]
```

Comment:
I like this pattern: https://github.com/mesosphere/dcos-cli/blob/master/cli/tests/unit/test_marathon_validation.py#L22-L31

Less error prone.